### DISCUSSION TITLE
Modifier to "Make Smaller" or "Make Larger" by some number of increments

### DISCUSSION THREAD

**Discussion by jonpierce** [rxhanson/Rectangle#286] [Ideas]
First off, as a long time Spectacle user who recently discovered it's no longer maintained, I was happy to stumble across Rectangle – what appears to be a more than worthy successor! Thank you!

I see you're not currently accepting new feature requests / suggestions, which is understandable. But I'm going to toss one out anyway, just in case it resonates.

I have a use case that requires me to adjust the size of my window by several increments of "Smaller" or "Larger". 

It would be cool if I could, say, use the "Make Smaller" or "Make Larger" key binding shortcut followed by a number to instantly adjust the size by that number of increments.

Anyway, just an idea. We know what those are worth. :)

**Comment by rxhanson** [comment]
Thanks for the kind words and the interesting idea!

I'm curious about your use case. Since you would know in advance how much smaller you'd want, you would probably have some idea as to what the right size would be, right? There's something about this that makes me feel like there would be another solution that might be a little more general but just as effective, since this specific solution is fairly specialized and not super reusable across the app. I like the idea though, since it's kinda vim-esque. Maybe there's other things that could be modeled with some vim convenience in mind.

**Comment by jonpierce** [comment]
My specific use case is an external display that recently developed a band of dead pixels that I've been trying to work around by resizing the windows on either side of it to fit. Obviously, this particular scenario is as much an edge case as you'll find. But I can imagine others where more customized window sizes / arrangements might be desired. I was able to achieve this through the "Make Smaller" / "Larger" key bindings but thought a modifier could make the process even more efficient. It's definitely not the only – nor necessarily best – option, though; for example, one could imagine allowing users to define custom slots, or regions, into which windows could be snapped (whether by drawing them on the screen, entering pixel dimensions, etc.). But the UX for that could also get unwieldy quickly.

**Comment by rxhanson** [comment]
Ah, I see. That is quite the edge case. The only other scenario where I think this type of behavior would be useful in Rectangle is if the move window in a direction actions were able to incrementally move the windows a small amount per keypress instead of snapping to the edge - a feature request that has come up. 

As it is, it's a little too hard to justify the time spent on specifically this scenario, but I would consider a pull request for it.

I did build a custom window shortcut builder UI for Rectangle, but as you mentioned, it was complex enough that it just didn't feel right adding it in. I ended up adding it only in my paid app [Hookshot](https://hookshot.app). 

I try to keep my open issues tidy, so I'm converting this to a discussion.
